rllib/tests/test_catalog.py

```diff
@@ -88,34 +88,48 @@ def test_custom_preprocessor(self):
     def test_default_models(self):
         ray.init(object_store_memory=1000 * 1024 * 1024)
 
-        for fw in framework_iterator(frameworks=("jax", "tf", "tf2", "torch")):
-            obs_space = Box(0, 1, shape=(3,), dtype=np.float32)
-            p1 = ModelCatalog.get_model_v2(
-                obs_space=obs_space,
-                action_space=Discrete(5),
-                num_outputs=5,
-                model_config={},
-                framework=fw,
-            )
-            self.assertTrue("FullyConnectedNetwork" in type(p1).__name__)
-            # Do a test forward pass.
-            obs = np.array([obs_space.sample()])
-            if fw == "torch":
-                obs = torch.from_numpy(obs)
-            out, state_outs = p1({"obs": obs})
-            self.assertTrue(out.shape == (1, 5))
-            self.assertTrue(state_outs == [])
-
-            # No Conv2Ds for JAX yet.
-            if fw != "jax":
-                p2 = ModelCatalog.get_model_v2(
-                    obs_space=Box(0, 1, shape=(84, 84, 3), dtype=np.float32),
-                    action_space=Discrete(5),
-                    num_outputs=5,
-                    model_config={},
-                    framework=fw,
-                )
-                self.assertTrue("VisionNetwork" in type(p2).__name__)
+        # Build test configs
```

Comment:
I'd suggest being more explicit here about each case so that maintenance of the test and expanding to would be easier for later:
One idea off the top of my head is to enumerate a few of these input / output shapes that require explicit testing. Then have a dictionary mapping frameworks to the supported cases.  For instance:
```
nested_case = {'input': gym.spaces.Dict(...), 'input_shape',: ... output_shape': ...}
fw_to_test_map = {'torch': [nested_case], ...}
```
Then loop through each framework, retrieve the test cases and check the assertions.